src/test/discourse-api/discourse-context.ts

```diff
@@ -200,17 +200,20 @@ export class DiscourseContext {
   }
 
   // XXX why not just chain the promises?
-  private async getCSRFAndCookieThen(callback: (csrf: string, cookie: string) => any): Promise<any> {
-      let res: IResponse = await rp({
-      uri: `${this.apiRoot}/session/csrf.json`,
-      json: true,
-      resolveWithFullResponse: true,
+  private getCSRFAndCookieThen(callback: (csrf: string, cookie: string) => any): Promise<any> {
+      return new Promise((resolve, reject) => {
```

Comment:
not working,
RP is not returning native promises, but Bluebird one,
used this to convert Bluebird to native.
Also, this code will be eliminated soon if i got it right from @helfer